generator/src/lib.rs

```diff
@@ -84,15 +84,20 @@ impl quote::ToTokens for CType {
 }
 
 fn parse_ctype(i: &str) -> IResult<&str, CType> {
-    (alt((value(CType::U64, tag("ULL")), value(CType::U32, tag("U"))))).parse(i)
+    (alt((
+        value(CType::U64, tag("ULL")),
+        value(CType::U32, tag("U")),
+        // success(CType::U32), // Only for `0x1234` hexadecimals without type-suffix
```

Comment:
Was mainly "experimenting" with removing the `Number` and `Hex` variants from `enum Constant` because this code already handles both, but they'd end up weirdly formatted.

We have logic that doesn't append the type (`ctype` parsing does it by default) and interleaves the number with `_`.

EDIT: Though it wouldn't look out of order to just delete the `0x...u32` suffix.

Reply:
I don't think the formatting of generated literals is very important. I guess it'd be nice to easily eyeball against the spec, but it doesn't affect downstream.